lib/cocoapods/installer.rb

```diff
@@ -318,6 +323,14 @@ def create_and_save_projects(pod_targets_to_generate, aggregate_targets_to_gener
       end
     end
 
+    def predictabilize_uuids
```

Comment:
nit: Can we have the functions pass in a list of projects. And should they also be private?

Reply:
What is the rationale for this, exactly? :generated_projects is an instance prop.

Reply:
Injecting parameters is generally a better practice and makes it much easier to reason about each property's state especially given that it's a class instance var not assigned in the initializer. This will also make refactoring much easier in the future.

Reply:
So why is there an instance property? Should I remove it?

Reply:
Ok, I see your point. I've made the methods receive the projects as a parameter. 

They are private because they are called from a private method. Do you think that's a good reason? I don't see any other reason to call them in this context and for other contexts they can use the generator classes.

Reply:
Ah they're already marked private, I missed the `private` above.